### PR TITLE
Refactor define_style_ensemble and unit tests

### PR DESCRIPTION
**Pull request by hnformentin**
**Issue**
Resolves partially #317

### CODE REVIEW

File: webviz_ert/controllers/response_correlation_controller.py [hunk 1]
```diff
@@ -466,6 +459,19 @@ def update_corr_param_resp(
         return corr_param_resp
 
 
+def define_style_ensemble(index: int, x_axis: pd.DataFrame) -> Dict:
```
Comment (frode-aarstad) [inline webviz_ert/controllers/response_correlation_controller.py:462]: perhaps a _ prefix if this is an internal function

**Review by valentin-krasontovitsch — APPROVED** [review]
besides my one question, this looks very nice!

**Review by DanSava — APPROVED** [review]
:+1:

File: tests/views/test_response_correlation.py [hunk 2]
```diff
@@ -158,3 +158,25 @@ def test_get_first_observation_x_invalid():
...
         _get_first_observation_x(df_observation)
+
+
+@pytest.mark.parametrize(
+    "index,expected_color", [(0, "rgba(56,108,176,0.8)"), (1, "rgba(127,201,127,0.8)")]
+)
+def test_define_style_ensemble_color(index, expected_color):
+    x_axis = pd.DataFrame([1])
+    style = _define_style_ensemble(index, x_axis)
+    assert style["line"]["color"] == expected_color
```
[the remark this answers is not in the document]
Comment (hnformentin) [reply tests/views/test_response_correlation.py:169]: Good point. When I print the variable `style` what I get is `{'mode': 'markers+lines', 'line': {'color': 'rgba(253,192,134,0.8)'}, 'marker': {'color': 'rgba(253,192,134,0.8)'}}`. So maybe in the test related to color, I can also test that the marker color matches with the line. Would that be good enough?

File: webviz_ert/controllers/response_correlation_controller.py [hunk 3]
```diff
@@ -466,6 +459,19 @@ def update_corr_param_resp(
...
+def _define_style_ensemble(index: int, x_axis: pd.DataFrame) -> Dict:
+    if x_axis is not None:
+        if str(x_axis[0]).isnumeric():
+            style = deepcopy(assets.ERTSTYLE["response-plot"]["response-index"])
+        else:
+            style = deepcopy(assets.ERTSTYLE["response-plot"]["response"])
+    ensemble_color = assets.get_color(index=index)
+    style.update({"marker": {"color": ensemble_color}})
+    style.update({"line": {"color": ensemble_color}})
+    print(style)
```
[the remark this answers is not in the document]
Comment (hnformentin) [reply webviz_ert/controllers/response_correlation_controller.py:471]: My bad, thanks!

File: tests/views/test_response_correlation.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (valentin-krasontovitsch) [reply tests/views/test_response_correlation.py:169]: sure, that sounds good I think!

**Comment by hnformentin** [comment]
I am messing up here...

**Comment by hnformentin** [comment]
ok, the problem was that I started to use `Oh-My-Zsh Git` and used the wrong alias to commit. I was using `gca` thinking it was only `git commit --amend`, bit `a` actually stands for all 🐱

**Pull request merged**